### PR TITLE
feature(microservices): update acceptable types in RabbitMq urls

### PR DESCRIPTION
**Pull request by capaio**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: 5587

## Does this PR introduce a breaking change?
```
[ ] Yes
[x ] No
```

## Other information

### CODE REVIEW

File: packages/microservices/interfaces/microservice-configuration.interface.ts [hunk 1]
```diff
@@ -119,7 +119,7 @@ export interface NatsOptions {
 export interface RmqOptions {
   transport?: Transport.RMQ;
   options?: {
-    urls?: string[];
+    urls?: any[];
```
Comment (capaio) [inline packages/microservices/interfaces/microservice-configuration.interface.ts:122]: If i put 
`urls?: string[] | RmqUrl[]` 
I get this error:
`error TS2322: Type 'string[] | RmqUrl[]' is not assignable to type 'string[]'`

Comment (galesky) [reply packages/microservices/interfaces/microservice-configuration.interface.ts:122]: In that case, wouldn't it be safer to change it to
```javascript
urls?: string[] | RmqUrl[]
```
Matching the changes you introduces here:
https://github.com/nestjs/nest/blob/b8c479ebe5705becbfe760110f54c7ccc8bf0bff/packages/microservices/client/client-rmq.ts#L31

Comment (capaio) [reply packages/microservices/interfaces/microservice-configuration.interface.ts:122]: changing to 
`urls?: string[] | RmqUrl[]`
I get this error on build:
`error TS2322: Type 'string[] | RmqUrl[]' is not assignable to type 'string[]'`

that's why I'm letting `any`

Comment (kamilmysliwiec) [reply packages/microservices/interfaces/microservice-configuration.interface.ts:122]: Can you paste a line here that leads to this error?

Comment (capaio) [reply packages/microservices/interfaces/microservice-configuration.interface.ts:122]: when RmqUrl is 
```
export interface RmqOptions {
  transport?: Transport.RMQ;
  options?: {
    urls: RmqUrl[] | string[];
    queue?: string;
    prefetchCount?: number;
    isGlobalPrefetchCount?: boolean;
    queueOptions?: any;
    socketOptions?: any;
    noAck?: boolean;
    serializer?: Serializer;
    deserializer?: Deserializer;
    replyQueue?: string;
  };
}
```
and I run 
`npm run build`

i get 

```
packages/microservices/server/server-rmq.ts(39,5): error TS2322: Type 'string[] | RmqUrl[]' is not assignable to type 'string[]'.
  Type 'RmqUrl[]' is not assignable to type 'string[]'.
    Type 'RmqUrl' is not assignable to type 'string'.
TypeScript: 1 semantic error
TypeScript: emit succeeded (with errors)
[10:10:21] 'microservices' errored after 2.52 s
[10:10:21] Error: TypeScript: Compilation failed
    at Output.mightFinish (/home/ccappai/projects/nest/node_modules/gulp-typescript/release/output.js:130:43)
    at applySourceMap.then.appliedSourceMap (/home/ccappai/projects/nest/node_modules/gulp-typescript/release/output.js:65:22)
    at process._tickCallback (internal/process/next_tick.js:68:7)

```

Comment (capaio) [reply packages/microservices/interfaces/microservice-configuration.interface.ts:122]: nevermind, i realized now what was my mistake, I'm pushing fix now

File: packages/microservices/server/server-rmq.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ import {
   OutgoingResponse,
 } from '../interfaces/packet.interface';
 import { Server } from './server';
+import {RmqUrl} from "../external/rmq-url.interface";
```
Comment (kamilmysliwiec) [inline packages/microservices/server/server-rmq.ts:23]:
```suggestion
import { RmqUrl } from "../external/rmq-url.interface";
```

File: packages/microservices/interfaces/microservice-configuration.interface.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { Server } from '../server/server';
 import { CustomTransportStrategy } from './custom-transport-strategy.interface';
 import { Deserializer } from './deserializer.interface';
 import { Serializer } from './serializer.interface';
+import {RmqUrl} from "../external/rmq-url.interface";
```
Comment (kamilmysliwiec) [inline packages/microservices/interfaces/microservice-configuration.interface.ts:15]:
```suggestion
import { RmqUrl } from "../external/rmq-url.interface";
```

File: packages/microservices/client/client-rmq.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ import {
 } from '../constants';
 import { ReadPacket, RmqOptions, WritePacket } from '../interfaces';
 import { ClientProxy } from './client-proxy';
+import {RmqUrl} from "../external/rmq-url.interface";
```
Comment (kamilmysliwiec) [inline packages/microservices/client/client-rmq.ts:20]:
```suggestion
import { RmqUrl } from "../external/rmq-url.interface";
```

**Comment by kamilmysliwiec** [comment]
Can you run the `format` script?

**Comment by capaio** [comment]
doing now

**Pull request merged**